ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/resources/arjunacore/XAResourceRecord.java

```diff
@@ -328,14 +333,18 @@ public int topLevelAbort()
 				}
                 catch(RuntimeException e)
                 {
+					// TODO: Is this in the right catch block?
```

Comment:
@jmfinelli please can you elaborate on the TODO? Does it need resolving?

Reply:
^^^^

Reply:
Please can you elaborate on why you think a RuntimeException shouldn't trigger this?

Reply:
Catching `RuntimeException` there is not the problem. What I think should be changed is the message with ID 160450. That message covers cases where `rollback` fails. In this catch block, it seems that ArjunaJTA is handling a failure of `end`.

Reply:
I see, it is within a call regarding abort though. Personally I expect I would leave it as is given it is in the abort path. If there was no connection to rollback at this point I hope I would feel differently.